File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
        l = 0
        r = len(nums)-1
        
        max_area = 0
        
        
        while l < r:
            
            area = min(nums[l], nums[r]) * (r - l)
            max_area = max(area, max_area)
            if nums[l] > nums[r]:
                r -= 1
            else:
                l += 1
        
        return max_area
        

File: 11-container-with-most-water/11-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    
    let l = 0;
    let r = height.length - 1;
    let area = 0;
    let max_area = 0;
    
    while (l < r) {
        area = Math.min(height[l], height[r]) * (r - l);
        max_area = Math.max(area, max_area);
        
        if (height[l] > height[r]) {
            r --;
        } else {
            l ++;
        }
    }
    
    return max_area;
    
};